### PR TITLE
Add changelog for RC 4.5.0

### PR DESCRIPTION
**Pull request by pzehner**
This PR aims to update the changelog for the 4.5.0 release candidate. It is based on the [changelog issue](https://github.com/kokkos/kokkos/issues/7183) and has been edited.
It also updates the readme for the latest version (strictly speaking, `4.5.00`).

- [ ] Remove empty sections

### CODE REVIEW

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:32]: This isn't user-visible.
```suggestion
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:34]: These aren't user-visible either.
```suggestion
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:36]: I would spit this in two entries. We could also just drop the second part since it's mostly not visible to the user.

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:43]:
```suggestion
* Fix race conditions for Threads backend on ARM architectures [\#7498](https://github.com/kokkos/kokkos/pull/7498)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:46]:
```suggestion
* Fix run time behavior when compiling with `-fvisibility-hidden` [\#7284](https://github.com/kokkos/kokkos/pull/7284)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:47]: Fix 
```suggestion
* Fix linking with CrayClang compiler [\#7341](https://github.com/kokkos/kokkos/pull/7341). 
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:50]:
```suggestion
* Allow Kokkos_ENABLE_ATOMICS_BYPASS to skip mutexes to remediate performance regression from 4.4 [\#7369](https://github.com/kokkos/kokkos/pull/7369)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:56]:
```suggestion
* Allow an arbitrary execution space instance to be used in `Kokkos::Graph::submit(...)` [\#7249](https://github.com/kokkos/kokkos/pull/7249)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:57]:
```suggestion
* Improve performance of Kokkos::sort when std::sort is used [\#7264](https://github.com/kokkos/kokkos/pull/7264)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:67]:
```suggestion
* Major refactoring removing `TriBITS` paths [\#6164](https://github.com/kokkos/kokkos/pull/6164)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:78]:
```suggestion
* Fix `deep_copy (serial_exec, dst, src)` with multiple host backends [\#7245](https://github.com/kokkos/kokkos/pull/7245)
```

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:83]:
```suggestion
* SYCL: Fix configuring without architecture flags [\#7303](https://github.com/kokkos/kokkos/pull/72303)
```

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:36]: Yeah, I thought so.

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:5]:
```suggestion
[Full Changelog](https://github.com/kokkos/kokkos/compare/4.4.00...4.5.0.)
```

Comment (dalg24) [inline CHANGELOG.md:5]:
```suggestion
[Full Changelog](https://github.com/kokkos/kokkos/compare/4.4.01...4.5.00)
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:10]:
```suggestion
* Support building with Run-Time Type Information (RTTI) disabled
```
No PR to refer in particular.  This has been implemented over many.

Comment (pzehner) [reply CHANGELOG.md:10]: Ok, noted

**Review by Rombur — APPROVED** [review]

File: CHANGELOG.md
Comment (tcclevenger) [inline CHANGELOG.md:37]: Should these sections be deleted before merging this PR?

File: CHANGELOG.md
Comment (tcclevenger) [inline CHANGELOG.md:82]: Same here

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:37]: Yes, this is a reminder for me.

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:82]: Idem.

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:25]: I would list this under bug fix.
IIRC this also affected the CUDA backend,

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:42]: Need to rephrase this one.  Should not be mentioning "BasicView".
I am tempted to say merge with the entry above and just say something like adding support for Clacc

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:69]: Tempted to move that one to build system updates.  Thoughts?

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:75]: Please collocate all the graph entries (don't merge them just put them next to each other)

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:55]: That one was back ported to 4.4.1 wasn't it?

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:50]:
```suggestion
* Fix compilation for `parallel_reduce` `MDRange` with `Dynamic` scheduling [\#7478](https://github.com/kokkos/kokkos/pull/7478)
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:46]: This does not really capture what changed as far as users are concerned.  What about
```suggestion
* Implement `Experimental::partition_space` to produce truly independent execution spaces [\#7287](https://github.com/kokkos/kokkos/pull/7287)
```

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:55]: Indeed.

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:25]: The same PR is already referenced in Bug fixes.

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:42]: Why not just mentioning "Fix OpenACC `MDRangePolicy` errors"? Besides, this PR is not related to Clacc.

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:75]: Done.

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:69]: The PR is mostly about CMake files changes, so I think it's appropriate to move it.

File: CHANGELOG.md
Comment (pzehner) [reply CHANGELOG.md:50]: Added.

File: CHANGELOG.md
Comment (dalg24) [reply CHANGELOG.md:42]: This was a toolchain bug, essentially we got UB when using collapse clauses and we implemented a workaround

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:11]:
```suggestion
* Add new `KOKKOS_RELOCATABLE_FUNCTION` function annotation macro [\#5993](https://github.com/kokkos/kokkos/pull/5993)
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:9]:
```suggestion
* SYCL backend graduated to production ready
* Introduce new `SequentialHostInit` view allocation property [\#7229](https://github.com/kokkos/kokkos/pull/7229) (backported in 4.4.01)
```

File: CHANGELOG.md
Comment (dalg24) [reply CHANGELOG.md:25]:
```suggestion
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:28]:
```suggestion
* Introduce new architecture option `AMD_GFX942_APU` for MI300A [\#7462](https://github.com/kokkos/kokkos/pull/7462)
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:51]:
```suggestion
* Fix race conditions on ARM architectures [\#7498](https://github.com/kokkos/kokkos/pull/7498)
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:74]:
```suggestion
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:76]:
```suggestion
* Allow an arbitrary execution space instance to be used in `Kokkos::Graph::submit()` [\#7249](https://github.com/kokkos/kokkos/pull/7249)
```

File: CHANGELOG.md
Comment (dalg24) [inline CHANGELOG.md:92]: * Deprecate `{OpenMP,HPX}::is_asynchronous()` [\#7322](https://github.com/kokkos/kokkos/pull/7322)

**Review by dalg24 — APPROVED** [review]

**Pull request merged**
